src/core/components/sidepanels/panels/theme-configuration/ThemeConfigPanel.tsx

```diff
@@ -0,0 +1,155 @@
+
+import { cn } from "../../../../functions/Functions.ts";
+import { BorderRadiusInput, FontSelector, ColorPickerInput } from "./index.ts";
+import { ChaiBuilderThemeOptions } from "../../../../types/chaiBuilderEditorProps.ts";
+import { useAtom } from "jotai";
+import { themeValuesAtom } from "../../../../atoms/theme.ts";
+
+interface ThemeConfigProps {
+  className?: string;
+}
+
+const defaultThemeStructure: ChaiBuilderThemeOptions = {
+  fontFamily: {},
+  borderRadius: {},
+  colors: [],
+};
+
+const ThemeConfigPanel: React.FC<ThemeConfigProps> = React.memo(({ className = "" }) => {
+  const [currentMode, setCurrentMode] = React.useState<"light" | "dark">("light");
+
+  const getThemeFromProps = useBuilderProp("themeOptions", (themeOptions: ChaiBuilderThemeOptions) => themeOptions);
+  const [themeValues, setThemeValues] = useAtom(themeValuesAtom);
+
+  const activeThemeOptions: ChaiBuilderThemeOptions = getThemeFromProps(defaultThemeStructure);
+
+  const { t } = useTranslation();
+
+  // Font update handler
+  const handleFontChange = (key: string, newValue: string) => {
+    setThemeValues((prev) => ({
+      ...prev,
+      fontFamily: {
+        ...prev.fontFamily,
+        [key]: newValue,
+      },
+    }));
+  };
+
+  // Border radius update handler
+  const handleBorderRadiusChange = (value: string) => {
+    setThemeValues((prev) => ({
+      ...prev,
+      borderRadius: `${value}rem`,
+    }));
+  };
+
+  // Color update handler
+  const handleColorChange = (key: string, newValue: string) => {
+    setThemeValues((prev) => ({
+      ...prev,
+      colors: {
+        ...prev.colors,
+        [key]: {
+          ...prev.colors[key],
+          [currentMode]: newValue,
+        },
+      },
+    }));
+  };
+
+  // Group rendering function
+  const renderColorGroup = (group: any) => (
+    <AccordionItem value={group.group} key={group.group}>
+      <AccordionTrigger className="hover:no-underline">
+        <div className="flex items-center space-x-2">
+          <span>{group.group}</span>
+        </div>
+      </AccordionTrigger>
+      <AccordionContent>
+        <div className="grid grid-cols-1 gap-4 bg-white">
+          {Object.entries(group.items).map(([key, value]: [string, any]) => {
+            const cssVariable = Object.keys(value)[0];
+            const themeColor = themeValues?.colors?.[key]?.[currentMode];
+
+            return (
+              <div key={key} className="space-y-2">
+                <Label>{key}</Label>
+                <ColorPickerInput
+                  value={themeColor || value[cssVariable]}
+                  onChange={(newValue: string) => handleColorChange(key, newValue)}
+                />
+              </div>
+            );
+          })}
+        </div>
+      </AccordionContent>
+    </AccordionItem>
+  );
+
+  return (
+    <div className={cn("h-full w-full space-y-6", className)}>
+      {/* Fonts Section */}
+      <div className="grid grid-cols-2 gap-4">
```

Comment:
what will happen if user wants to disable fonts
with `fontFamily: false`

Reply:
Adding a boolean to show only if keys are present.